src/Arcus.Templates.AzureFunctions.Http/AzureFunction.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Arcus.WebApi.Logging.Correlation;
+using GuardNet;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+using JsonSerializer = Newtonsoft.Json.JsonSerializer;
+
+namespace Arcus.Templates.AzureFunctions.Http
+{
+    /// <summary>
+    /// Represents the base HTTP web API functionality for an Azure Function implementation.
+    /// </summary>
+    public abstract class AzureFunction
+    {
+        private readonly HttpCorrelation _httpCorrelation;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureFunction"/> class.
+        /// </summary>
+        /// <param name="httpCorrelation">The correlation service to provide information of related requests.</param>
+        /// <param name="logger">The logger instance to write diagnostic messages throughout the execution of the HTTP trigger.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="httpCorrelation"/> is <c>null</c></exception>
+        protected AzureFunction(HttpCorrelation httpCorrelation, ILogger logger)
+        {
+            Guard.NotNull(httpCorrelation, nameof(httpCorrelation), "Requires an HTTP correlation instance");
+            _httpCorrelation = httpCorrelation;
+
+            Logger = logger ?? NullLogger.Instance;
+        }
+
+        static AzureFunction()
```

Comment:
Hm I've never seen this before, is there a reason why we are adding this?

Reply:
Well, the JSON serializer settings/options are stuff that can be created once. If it were non-static, it would have to be created for each implementation of this class; now only once for all implementations.
But I see your point. Maybe this is taking things a bit too far.

Reply:
I'll move this to the general ctor. May not be worth the obscurity that comes with it.